ArborX neighborlist interface compile error

Nightly build fails using ArborX master

```
/__w/Cabana/Cabana/core/unit_test/tstNeighborListArborX.hpp:96:66:   required from here
/github/home/arborx/include/ArborX/details/ArborX_Callbacks.hpp:90:9: error: no type named 'Tag' in 'using Predicate = using value_type = std::decay_t<Kokkos::nonesuch>' {aka 'std::decay<Kokkos::nonesuch>::type'}
   90 |   using PredicateTag = typename Predicate::Tag;
      |         ^~~~~~~~~~~~
/github/home/arborx/include/ArborX/details/ArborX_Callbacks.hpp:99:55: error: no type named 'Tag' in 'using value_type = std::decay_t<Kokkos::nonesuch>' {aka 'std::decay<Kokkos::nonesuch>::type'}
   99 |   static_assert(is_valid_predicate_tag<PredicateTag>::value &&
      |                                                       ^~~~~
```

--- a/core/src/Cabana_Experimental_NeighborList.hpp
+++ b/core/src/Cabana_Experimental_NeighborList.hpp
@@ -96,24 +96,26 @@ struct AccessTraits<Slice, PrimitivesTag,
     }
 };
 //! Neighbor access trait.
-template <typename SliceLike>
-struct AccessTraits<SliceLike, PredicatesTag>
+template <typename Slice>
+struct AccessTraits<Cabana::Experimental::Impl::SubsliceAndRadius<Slice>,
+                    PredicatesTag>
 {
+    using Self = Cabana::Experimental::Impl::SubsliceAndRadius<Slice>;
     //! Kokkos memory space.
-    using memory_space = typename SliceLike::memory_space;
+    using memory_space = typename Self::memory_space;
     //! Size type.
-    using size_type = typename SliceLike::size_type;
+    using size_type = typename Self::size_type;
     //! Get number of particles.
-    static KOKKOS_FUNCTION size_type size( SliceLike const& x )
+    static KOKKOS_FUNCTION size_type size( Self const& x )
     {
         return x.last - x.first;
     }
     //! Get the particle at the index.
-    static KOKKOS_FUNCTION auto get( SliceLike const& x, size_type i )
+    static KOKKOS_FUNCTION auto get( Self const& x, size_type i )
     {
         assert( i < size( x ) );
         auto const point =
-            AccessTraits<typename SliceLike::slice_type, PrimitivesTag>::get(
+            AccessTraits<typename Self::slice_type, PrimitivesTag>::get(
                 x.slice, x.first + i );
         return attach( intersects( Sphere{ point, x.radius } ), (int)i );
     }

--- a/core/src/Cabana_Experimental_NeighborList.hpp
+++ b/core/src/Cabana_Experimental_NeighborList.hpp
@@ -415,7 +415,7 @@ auto make2DNeighborList( ExecutionSpace space, Tag,
         Impl::makePredicates( coordinate_slice, first, last, radius );
 
     auto const n_queries =
-        ArborX::AccessTraits<decltype( predicates ),
+        ArborX::AccessTraits<std::remove_const_t<decltype( predicates )>,
                              ArborX::PredicatesTag>::size( predicates );
 
     Kokkos::View<int**, memory_space> neighbors;

--- a/core/src/Cabana_Experimental_NeighborList.hpp
+++ b/core/src/Cabana_Experimental_NeighborList.hpp
@@ -100,22 +100,23 @@ template <typename Slice>
 struct AccessTraits<Cabana::Experimental::Impl::SubsliceAndRadius<Slice>,
                     PredicatesTag>
 {
-    using Self = Cabana::Experimental::Impl::SubsliceAndRadius<Slice>;
+    //! Slice wrapper with partial range and radius information.
+    using SliceLike = Cabana::Experimental::Impl::SubsliceAndRadius<Slice>;
     //! Kokkos memory space.
-    using memory_space = typename Self::memory_space;
+    using memory_space = typename SliceLike::memory_space;
     //! Size type.
-    using size_type = typename Self::size_type;
+    using size_type = typename SliceLike::size_type;
     //! Get number of particles.
-    static KOKKOS_FUNCTION size_type size( Self const& x )
+    static KOKKOS_FUNCTION size_type size( SliceLike const& x )
     {
         return x.last - x.first;
     }
     //! Get the particle at the index.
-    static KOKKOS_FUNCTION auto get( Self const& x, size_type i )
+    static KOKKOS_FUNCTION auto get( SliceLike const& x, size_type i )
     {
         assert( i < size( x ) );
         auto const point =
-            AccessTraits<typename Self::slice_type, PrimitivesTag>::get(
+            AccessTraits<typename SliceLike::slice_type, PrimitivesTag>::get(
                 x.slice, x.first + i );
         return attach( intersects( Sphere{ point, x.radius } ), (int)i );
     }

